tools/wptrunner/wptrunner/executors/executorwebdriver.py

```diff
@@ -976,6 +995,12 @@ def on_environment_change(self, new_environment):
     def do_test(self, test):
         url = self.test_url(test)
 
+        if self.require_webdriver_bidi and not self.protocol.enable_bidi:
+            # WebDriver BiDi is required for the test, but the test is not an implementation of
+            # WebDriverBidiProtocol. The test cannot be run. Provide a meaningful error message in
+            # this case.
+            return (test.make_result("ERROR", "test requires enabled WebDriver BiDi"), [])
+
```

Comment:
I wonder if that's the right result; I think `SKIP` may make more sense, c.f. https://github.com/web-platform-tests/wpt/blob/86eea000de7bd4b15640fcb75a5568de5e243891/tools/wptrunner/wptrunner/wptrunner.py#L266-L277

I also wonder if we should actually put this in `run_test` rather than `do_test`? We don't need to change environment and everything if we're just going to skip it.